src/shared/sam/cli/samCliInvokerUtils.ts

```diff
@@ -55,6 +55,42 @@ export function logAndThrowIfUnexpectedExitCode(processResult: ChildProcessResul
             message = processResult.error.message
         }
     }
+    const usefulErrors = collectAcceptedErrorMessages(processResult.stderr).join('\n')
+    throw makeUnexpectedExitCodeError(message ?? usefulErrors)
+}
+
+/**
+ * Collect known errors messages from a sam error message
+ * that may have multiple errors in one message.
+ * @param errors A string that can have multiple error messages
+ */
+export function collectAcceptedErrorMessages(errorMessage: string): string[] {
+    const errors = errorMessage.split('\n')
+    const shouldCollectFuncs = [_startsWithEscapeSequence, _startsWithError]
+    return errors.filter(error => {
+        return shouldCollectFuncs.some(shouldCollect => {
+            return shouldCollect(error)
+        })
+    })
+}
+
+/**
+ * Returns true if text starts with an escape
+ * sequence with one of the accepted sequences.
```

Comment:
nit: We mention accepted sequences but don't say what they are or how they are derived. Perhaps move `acceptedSequences` outside of the function body.